sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -510,11 +510,15 @@ rowFormat
     ;
 
 tableIdentifier
-    : (db=identifier '.')? table=identifier
+    : (db=identifierPart '.')? table=identifierPart
     ;
 
 functionIdentifier
-    : (db=identifier '.')? function=identifier
+    : (db=identifierPart '.')? function=identifierPart
+    ;
+
+identifierPart
+    : identifier | BIGINT_LITERAL | SMALLINT_LITERAL | TINYINT_LITERAL | BYTELENGTH_LITERAL
```

Comment:
`INTEGER_VALUE`?

Reply:
`INTEGER_VALUE` should not be here because it cannot be an identifier.

Reply:
But the rule in `validateName` allows a name consisting of numbers.